test/e2e/specs/wp-calypso-gutenberg-checkout-spec.js

```diff
@@ -0,0 +1,260 @@
+		step( 'Can log in', async function () {
+			this.timeout( mochaTimeOut * 12 );
+			const loginFlow = new LoginFlow( driver, 'gutenbergSimpleSiteFreePlanUser' );
+			return await loginFlow.loginAndStartNewPost( null, true );
+		} );
+
+		step( 'We can set the sandbox cookie for payments', async function () {
+			const wPHomePage = await WPHomePage.Visit( driver );
+			await wPHomePage.checkURL( locale );
+			await wPHomePage.setSandboxModeForPayments( sandboxCookieValue );
+			await wPHomePage.setCurrencyForPayments( currencyValue );
+			return await driver.navigate().back();
+		} );
+
+		step( 'Can insert the premium block', async function () {
+			const gEditorComponent = await GutenbergEditorComponent.Expect( driver );
+			await gEditorComponent.addBlock( 'Simple Payments' );
+			return await driverHelper.waitTillPresentAndDisplayed(
+				driver,
+				By.css( '.wp-block-jetpack-simple-payments' )
+			);
+		} );
+
+		step( 'Can click Upgrade button on premium block', async function () {
+			const gEditorComponent = await GutenbergEditorComponent.Expect( driver );
+			await gEditorComponent.ensureSaved();
+			await gEditorComponent.clickUpgradeOnPremiumBlock();
+			return await driver.switchTo().defaultContent();
+		} );
+
+		step( 'Can view checkout modal', async function () {
+			editorUrl = await driver.executeScript( 'return window.location.href' );
+			await driverHelper.waitTillPresentAndDisplayed( driver, By.css( '.editor-checkout-modal' ) );
+			const compositeCheckoutIsPresent = await driverHelper.isElementPresent(
+				driver,
+				By.css( '.editor-checkout-modal' )
+			);
+			assert.strictEqual(
+				compositeCheckoutIsPresent,
+				true,
+				'The in-editor checkout is not present'
+			);
+		} );
+	} );
+
+	describe( 'Has correct plan details', function () {
+		step( 'Contains Premium Plan', async function () {
+			const securePaymentComponent = await SecurePaymentComponent.Expect( driver );
+			const checkoutContainsPremiumPlan = await securePaymentComponent.containsPremiumPlan();
+			assert.strictEqual(
+				checkoutContainsPremiumPlan,
+				true,
+				'The in-editor checkout does not contain the expected Premium Plan'
+			);
+		} );
+
+		step( 'Can change plan length', async function () {
+			const securePaymentComponent = await SecurePaymentComponent.Expect( driver );
+			const originalCartAmount = await securePaymentComponent.cartTotalAmount();
+			await driverHelper.waitTillPresentAndDisplayed(
+				driver,
+				By.css( '.wp-checkout__review-order-step .checkout-step__edit-button' )
+			);
+			await driverHelper.clickWhenClickable(
+				driver,
+				By.css( '.wp-checkout__review-order-step .checkout-step__edit-button' )
+			);
+			const twoYearPlan = await driver.findElement(
+				By.css( '.wp-checkout__review-order-step ul > li:nth-child(2) > div' )
+			);
+			await twoYearPlan.click();
+			// TODO: This seems flakey, need to think of a way around this.
+			await driver.sleep( 1500 );
```

Comment:
I see your TODO. What's the reason for the sleep? Won't the `waitTilPresentAndDisplayed` do that for you? (I assume for some reason it won't which is why this is here, but I had to ask.)